modules/client/js/src/main/scala/com/azavea/stac4s/api/client/SearchFilters.scala

```diff
@@ -14,10 +14,10 @@ case class SearchFilters(
     bbox: Option[Bbox] = None,
     datetime: Option[TemporalExtent] = None,
     intersects: Option[Geometry] = None,
-    collections: List[String] = Nil,
-    items: List[String] = Nil,
+    collections: Option[List[String]] = None,
```

Comment:
`Option[List[String]]` is the saddest type in the universe :( -- do you think it's correct that if this is present, the list should not be empty? Right now it's "zero or one zero or many"s

Reply:
@jisantuc well, I need to erase nulls, I realized that it can be extremely convenient not to send nulls. I.e. with the `query` parameter - if it is set to `{}` it may confuse servers that don't implement it.

I don't have strong opinions about Option[List] collections and items (I can double check that), but I have pretty strong opinions about `query`, which is `Option[Map[String, List[Query]]]`. I faced issues with https://eod-catalog-svc-prod.astraea.earth/

Reply:
Moved the solution into the codecs area!